### PR TITLE
Preparing for dockerization

### PR DESCRIPTION
**Pull request by carueda**
@dwcaress Just an initial PR to start integrating the dockerization of the system.

Right now, this PR just adds a subdirectory `docker` with a Dockerfile and a readme. There're pending aspects to be defined (hopefully soon) but I'd like for us to have a basis moving forward. 

Most immediate aspects to consider include:

- I'm currently setting up the automatic image build and publication upon a push to the master branch.  This uses a mechanism defined on Docker Hub. 
- Publication of the image.  Currently done under the MBARI org at Docker Hub: https://hub.docker.com/r/mbari/mbsystem .
- In general, we need to define the various relevant "spaces." For example, maybe you want the image to be published somewhere else?

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
+FROM centos:7
```
Comment (schwehr) [inline docker/Dockerfile]: Why centos specific stuff in `docker/Dockerfile`?

**Comment by schwehr** [comment]
Having the latest pushed to DockerHub sounds awesome!

**Comment by carueda** [comment]
(Sorry, not sure what I clicked here on the github interface but seems like I mistakenly "resolved" your comment  and I'm not seeing an "undo") ... anyway:

> Why centos specific stuff in docker/Dockerfile?

It's just the base image we are going to use at least for the time being.

**Comment by carueda** [comment]
Also, after just talking with Dave, I'm going to continue the preparations on a separate repo for now.

File: docker/Dockerfile (hunk 1 above)
Comment (carueda) [reply docker/Dockerfile]: It's just the base image we are going to use at least for the time being.

**Pull request merged**